packages/clerk-js/src/ui/components/UserProfile/Web3Section.tsx

```diff
@@ -3,10 +3,35 @@ import { useUser } from '@clerk/shared/react';
 import { Flex, Image, localizationKeys } from '../../customizables';
 import { ProfileSection, ThreeDotsMenu } from '../../elements';
 import { Action } from '../../elements/Action';
+import { useActionContext } from '../../elements/Action/ActionRoot';
 import { useEnabledThirdPartyProviders } from '../../hooks';
 import type { PropsOfComponent } from '../../styledSystem';
 import { RemoveWeb3WalletForm } from './RemoveResourceForm';
-import { Web3Form } from './Web3Form';
+import { AddWeb3Wallet } from './Web3Form';
+
+type RemoveWeb3WalletScreenProps = { walletId: string };
+const RemoveWeb3WalletScreen = (props: RemoveWeb3WalletScreenProps) => {
+  const { close } = useActionContext();
+  return (
+    <RemoveWeb3WalletForm
+      onSuccess={close}
+      onReset={close}
+      {...props}
+    />
+  );
+};
+
+type Web3WalletScreenProps = { walletId?: string };
+const Web3WalletScreen = (props: Web3WalletScreenProps) => {
```

Comment:
🙃 Does this benefit us ?
```suggestion
const Web3WalletScreen = (props: ComponentProp<typeof AddWeb3Wallet>) => {
```